controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -2164,6 +2164,7 @@ int bml_internal::get_wifi_credentials(int vap_id, char *ssid, char *pass, int *
     //clear the memory with password in it.
     memset(&sWifiCredentials.pass, 0, sizeof(sWifiCredentials.pass));
     //memset might be optimized and compiler might not set it 0 if its not used after memset
+    // cppcheck-suppress selfAssignment
     *(volatile char *)sWifiCredentials.pass = *(volatile char *)sWifiCredentials.pass;
```

Comment:
I think this is a terrible hack. A short google search says that `memset_s` will not be optimized out - https://www.cryptologie.net/article/419/zeroing-memory-compiler-optimizations-and-memset_s/
I would really prefer to fix the code to use memset_s with a comment instead of this hack.
In general, I prefer not to add the `// cppcheck-suppress ` comments in our production code.

Reply:
memset_s is C11, not C++. cppreference [advises to use std::fill with volatile instead](https://en.cppreference.com/w/cpp/string/byte/memset#Notes). I like that solution a lot better.

Reply:
I find the cppreference advice confusing. Does the pointer itself really need to be volatile?

Said differently: AFAIK calling std:fill with a `volatile char *p` should be fine. But does it really guarantee that it won't be optimized by any C++11-compliant compiler?
Or should we call it with both the pointer and the pointee volatile? (`volatile char *p volatile`) ?

Reply:
Indeed, they mean with `volatile char* p`. "with volatile pointers" means "with pointers to volatile data". An actual `char * volatile p` is pretty useless, and casting to it is not even possible.